src/components/ErrorBoundary/index.native.tsx

```diff
@@ -13,4 +13,11 @@ BaseErrorBoundary.defaultProps.logError = (errorMessage, error, errorInfo) => {
     crashlytics().recordError(error);
 };
 
-export default BaseErrorBoundary;
+export default (props: BaseErrorBoundaryProps) => {
```

Comment:
Let's create as a own component:

```ts
function ErrorBoundary(props: BaseErrorBoundaryProps) {
    return (
        <BaseErrorBoundary
            {...props}
            logError={logError}
        />
    );
}

ErrorBoundary.displayName = 'ErrorBoundary';

export default ErrorBoundary;
```

In the example I inverted `logError` and `{...props}` to make sure `logError` is not overridden by whoever uses this component.

Reply:
I agree that we should prevent such an approach, but in this case, if we change the logic it would not be possible to overwrite logError function with the prop

Reply:
I see that we do not use this logic to overwrite